### PR TITLE
[LUCENE-8075] Possible null pointer dereference in core/src/java/org/…

### PR DESCRIPTION
**Pull request by imgpulak**
…apache/lucene/codecs/blocktree/IntersectTermsEnum.java

Fix it

### CODE REVIEW

**Comment by jpountz** [comment]
Actually I wonder that we should instead remove the `if (fr.index == null)` branch. @mikemccand We do not seem to use the ability to skip the load of the terms index anymore?

File: lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java [hunk 1]
```diff
@@ -106,37 +106,37 @@ public IntersectTermsEnum(FieldReader fr, Automaton automaton, RunAutomaton runA
     if (fr.index == null) {
       fstReader = null;
```
[the remark this answers is not in the document]
Comment (jpountz) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: My point was that it seems that fr.index can never be null. So we will never enter this branch.

[the remark this answers is not in the document]
Comment (imgpulak) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: Okay. In that case shall we that piece of code? @jpountz

[the remark this answers is not in the document]
Comment (jpountz) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: This is why I pinged @mikemccand to know whether the fact that the index is always loaded is accidental or desired.

[the remark this answers is not in the document]
Comment (imgpulak) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: Okay. Now I understand. Thanks! Let me know what we are going to do. I will make changes accordingly.

[the remark this answers is not in the document]
Comment (imgpulak) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: Hi @jpountz, 

May I get the design documents of Lucene?

Thanks, 
Pulak

[the remark this answers is not in the document]
Comment (jpountz) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: All documentation is in javadocs, we don't have other documentation.

[the remark this answers is not in the document]
Comment (imgpulak) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: Thanks. That works for me.

[the remark this answers is not in the document]
Comment (imgpulak) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: @jpountz Any update here?

[the remark this answers is not in the document]
Comment (imgpulak) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: @jpountz, @mikemccand any update here?

[the remark this answers is not in the document]
Comment (mikemccand) [reply lucene/core/src/java/org/apache/lucene/codecs/blocktree/IntersectTermsEnum.java]: Sorry for the slow respons ehere @imgpulak and @jpountz but Adrien is right: `fr.index` can never be null anymore.  So I think we should change the code to `assert` it's never null and only do the `else` clause of the current `if` statement?

**Comment by imgpulak** [comment]
@ jpountz @mikemccand I have updated as per your comment. Please review the same. 
Thanks !

**Pull request closed without merging**